src/main/java/ro/hiringsystem/service/JobApplicationService.java

```diff
@@ -0,0 +1,26 @@
+package ro.hiringsystem.service;
+
+import ro.hiringsystem.model.dto.JobApplicationDto;
+import ro.hiringsystem.model.dto.JobDto;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public interface JobApplicationService {
+
+    JobApplicationDto getById(UUID id);
+
+    Map<UUID, JobApplicationDto> getAllFromMap();
+
+    void addAllFromGivenMap(Map<UUID, JobApplicationDto> jobApplicationMap);
+
+    void add(JobApplicationDto jobApplicationDto);
+
+    void removeElementById(UUID id);
+
+    void updateElementById(JobApplicationDto newJobApplicationDto);
```

Comment:
This method has been renamed to saveElement in the Spring Security pull request and saves the user to database no matter if it was already in there or not, please update it here as well so there are no merging conflicts